frontend/src/modules/notification/models.ts

```diff
@@ -0,0 +1,37 @@
+import { ReactNode } from "react"
+import { AnyAction } from "redux"
+
+export const ALERT = 'ALERT'
+export const ALERT_REMOVE = `${ALERT}.REMOVE`
+
+export enum AlertType {
+    SUCCESS = 'success',
+    INFO = 'info',
+    WARNING = 'warning',
+    ERROR = 'error'
+}
+
+export interface AlertProps {
+    type: AlertType,
+    description: ReactNode
+    title?: string,
+    error?: Error
+}
+
+export type AlertID = string
```

Comment:
I like using a string for an ID. For network errors (failed api calls) we could use the URL as an id string. If the same error occurs 500 times, it would ensure that the existing notification is replaced by the new notification, rather than generating 500 alert components. 👍